code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/InternalMessagingUtils.java

```diff
@@ -560,6 +562,83 @@ static String getSharedStateEcid(final Map<String, Object> edgeIdentityState) {
         return DataReader.optString(ecidMap, MessagingConstants.SharedState.EdgeIdentity.ID, null);
     }
 
+    /**
+     * Determines if the provided push token should be synced to Adobe Journey Optimizer via an Edge
+     * network request.
+     *
+     * @param configSharedState A {@link Map} containing the configuration shared state
+     * @param newPushToken A {@code String} containing the push token to be synced
+     * @return {@code boolean} indicating if the push token should be synced
+     */
+    static boolean shouldSyncPushToken(
+            @Nullable final Map<String, Object> configSharedState,
+            @Nullable final String newPushToken) {
+        if (StringUtils.isNullOrEmpty(newPushToken)) {
+            Log.debug(
+                    MessagingConstants.LOG_TAG,
+                    "shouldSyncPushToken",
+                    "New push token is null or empty, push token will not be synced.");
+            return false;
```

Comment:
In Messaging, if the new push token is null/empty while the persisted token is not, do you need to sync the empty push token to Profile?

Reply:
good callout, i'll still sync a token if it is empty.

Reply:
we ignore push sync events which do not have a valid push token on android and iOS. this is the existing behavior but i think it would be good to discuss if an empty token should still be synced.

Reply:
Yes, we can discuss if null tokens are allowed and will clear the local token. If we decide to make that change we can take it up in another PR.